### PR TITLE
faq, toc: add folder for FAQs and rename accordingly

### PR DESCRIPTION
**Pull request by dcalvin**
This PR added a FAQ folder and re-organized FAQ files accordingly, to be consistent with docs. Some further updates include:

- Move troubleshooting files of DM, Lightning to How-to
- Update TOC accordingly

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -172,13 +172,16 @@
...
         - [从 Amazon Aurora MySQL 迁移数据](dev/how-to/migrate/from-aurora.md)
+      - [Data Migration 故障诊断](dev/how-to/troubleshoot/data-migration.md)
     + TiDB-Lightning
       - [概述](dev/reference/tools/tidb-lightning/overview.md)
       - [部署执行](dev/reference/tools/tidb-lightning/deployment.md)
       - [断点续传](dev/reference/tools/tidb-lightning/checkpoints.md)
       - [表库过滤](dev/reference/tools/tidb-lightning/filter.md)
       - [CSV 支持](dev/reference/tools/tidb-lightning/csv.md)
       - [监控告警](dev/reference/tools/tidb-lightning/monitor.md)
+      - [TiDB-Lightning 故障诊断](dev/how-to/troubleshoot/tidb-lightning.md)
```
Comment (lilin90) [inline TOC.md]:
```suggestion
      - [故障诊断](dev/how-to/troubleshoot/tidb-lightning.md)
```

File: TOC.md [hunk 2]
```diff
@@ -172,13 +172,16 @@
...
+      - [TiDB-Lightning FAQ](faq/tidb-lightning.md)
```
Comment (lilin90) [inline TOC.md]:
```suggestion
      - [FAQ](faq/tidb-lightning.md)
```

File: TOC.md [hunk 3]
```diff
@@ -191,9 +194,9 @@
     - [TiDB Controller](dev/reference/tools/tidb-control.md)
     - [工具下载](dev/reference/tools/download.md)
 + 常见问题 (FAQ)
-  - [TiDB FAQ](FAQ.md)
-  - [TiDB-Lightning FAQ](tools/lightning/faq.md)
-  - [升级 FAQ](op-guide/upgrade-faq.md)
+  - [TiDB FAQ](faq/tidb.md)
```
Comment (lilin90) [inline TOC.md]: This renaming affects the following links:

- Two links in [this file](https://github.com/pingcap/docs-cn/blob/master/dev/reference/error-codes.md)
- One link in [this file](https://github.com/pingcap/docs-cn/blob/master/dev/reference/tools/tidb-lightning/deployment.md)

Please update them.

File: dev/reference/error-codes.md [hunk 4]
```diff
@@ -23,8 +23,8 @@ TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样
 | 9004 | 当数据库上承载的业务存在大量的事务冲突时，会遇到这种错误，请检查业务代码 |
 | 9005 | 某个 Raft Group 不可用，如副本数目不足，出现在 TiKV 比较繁忙或者是 TiKV 节点停机的时候，请检查 TiKV Server 状态/监控/日志 |
 | 9006 | GC Life Time 间隔时间过短，长事务本应读到的数据可能被清理了，应增加 GC Life Time |
-| 9500 | 单个事务过大，原因及解决方法请参考[这里](/FAQ.md#出现-transaction-too-large-报错怎么办) |
+| 9500 | 单个事务过大，原因及解决方法请参考[这里](faq/tidb.md#出现-transaction-too-large-报错怎么办) |
```
Comment (dcalvin) [inline dev/reference/error-codes.md]:
```suggestion
| 9500 | 单个事务过大，原因及解决方法请参考[这里](/faq/tidb.md#出现-transaction-too-large-报错怎么办) |
```

File: dev/reference/error-codes.md [hunk 5]
```diff
@@ -23,8 +23,8 @@ TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样
...
 
 ## 故障诊断
 
-参见[故障诊断文档](/dev/how-to/troubleshoot/cluster-setup.md)以及 [FAQ](/FAQ.md)。
+参见[故障诊断文档](/dev/how-to/troubleshoot/cluster-setup.md)以及 [FAQ](faq/tidb.md)。
```
Comment (dcalvin) [inline dev/reference/error-codes.md]:
```suggestion
参见[故障诊断文档](/dev/how-to/troubleshoot/cluster-setup.md)以及 [FAQ](/faq/tidb.md)。
```

**Pull request merged**